### PR TITLE
Implement support for the IRCv3 SETNAME specification.

### PR DESCRIPTION
**Pull request by SadieCat**
Exactly what it says on the tin.

### CODE REVIEW

File: src/common/userlist.c [hunk 1]
```diff
@@ -130,8 +130,12 @@ userlist_add_hostname (struct session *sess, char *nick, char *hostname,
 			g_free (user->hostname);
 			user->hostname = g_strdup (hostname);
 		}
-		if (!user->realname && realname && *realname)
+		if (realname && (!user->realname || strcmp(user->realname, realname)))
```
Comment (TingPing) [inline src/common/userlist.c:133]:
```suggestion
        if (realname && g_strcmp0 (user->realname, realname) != 0)
```

Comment (TingPing) [reply src/common/userlist.c:133]: I'm not sure if the empty string check is necessarily what you expect, but it keeps previous behavior.

File: src/common/userlist.c [hunk 2]
```diff
@@ -130,8 +130,12 @@ userlist_add_hostname (struct session *sess, char *nick, char *hostname,
...
+		{
+			do_rehash = TRUE;
```
Comment (TingPing) [inline src/common/userlist.c:135]: Is `do_rehash` needed? That is for elements that show up in the userlist.

Comment (SadieCat) [reply src/common/userlist.c:135]: I was going off nearby code so I thought it did but now I check it doesn't. Removed.

**Pull request merged**